### PR TITLE
Refresh Terraform related documentation

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This pull request refreshes the Terraform related documentation in this project. This includes updating instructions to represent the changes that occurred from the Terraform 0.12->1.0 upgrade process as well as just updating the documentation a bit in general.

## 💭 Motivation and context ##

The directions in this repository were never updated to reflect the changes made in the Terraform 0.12->1.0 upgrade and the documentation is a bit dated in formatting in general. Since I'm here doing some freshening up anyway I thought it would be worth it to fix.

## 🧪 Testing ##

Automated tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

File: terraform/variables.tf [hunk 1]
```diff
@@ -206,7 +206,7 @@ variable "assessment_data_filename" {
 
 variable "assessment_data_import_lambda_s3_bucket" {
   type        = string
-  description = "The name of the bucket where the assessment data import Lambda function can be found.  This bucket should be created with https://github.com/cisagov/assessment-data-import-terraform.  Note that in production terraform workspaces, the string '-production' will be appended to the bucket name.  In non-production workspaces, '-<workspace_name>' will be appended to the bucket name."
+  description = "The name of the bucket where the assessment data import Lambda function can be found.  This bucket should be created with the cisagov/assessment-data-import-terraform project.  Note that in production terraform workspaces, the string '-production' will be appended to the bucket name.  In non-production workspaces, '-<workspace_name>' will be appended to the bucket name."
```
Comment (dav3r) [inline terraform/variables.tf:209]: Keep the capitalization train rollin':
```suggestion
  description = "The name of the bucket where the assessment data import Lambda function can be found.  This bucket should be created with the cisagov/assessment-data-import-terraform project.  Note that in production Terraform workspaces, the string '-production' will be appended to the bucket name.  In non-production workspaces, '-<workspace_name>' will be appended to the bucket name."
```

File: terraform/variables.tf [hunk 2]
```diff
@@ -274,7 +274,7 @@ variable "findings_data_save_succeeded" {
 
 variable "findings_data_import_lambda_s3_bucket" {
   type        = string
-  description = "The name of the bucket where the findings data import Lambda function can be found.  This bucket should be created with https://github.com/cisagov/findings-data-import-terraform.  Note that in production terraform workspaces, the string '-production' will be appended to the bucket name.  In non-production workspaces, '-<workspace_name>' will be appended to the bucket name."
+  description = "The name of the bucket where the findings data import Lambda function can be found.  This bucket should be created with the cisagov/findings-data-import-terraform project.  Note that in production terraform workspaces, the string '-production' will be appended to the bucket name.  In non-production workspaces, '-<workspace_name>' will be appended to the bucket name."
```
Comment (dav3r) [inline terraform/variables.tf:277]: Keep the capitalization train rollin':
```suggestion
  description = "The name of the bucket where the findings data import Lambda function can be found.  This bucket should be created with the cisagov/findings-data-import-terraform project.  Note that in production Terraform workspaces, the string '-production' will be appended to the bucket name.  In non-production workspaces, '-<workspace_name>' will be appended to the bucket name."
```

**Review by dav3r — APPROVED** [review]
Good stuff as always! 💎

**Review by jsf9k — APPROVED** [review]
Cleanup on aisle three!

File: terraform/variables.tf (hunk 1 above)
Comment (mcdonnnj) [reply terraform/variables.tf:209]: Please see [b1bf5f0](https://github.com/cisagov/cyhy_amis/pull/361/commits/b1bf5f0a3dd5f7ff6e973ec2c15582f4a3cd2b56).

File: terraform/variables.tf (hunk 2 above)
Comment (mcdonnnj) [reply terraform/variables.tf:277]: Please see [b1bf5f0](https://github.com/cisagov/cyhy_amis/pull/361/commits/b1bf5f0a3dd5f7ff6e973ec2c15582f4a3cd2b56).

**Pull request merged**